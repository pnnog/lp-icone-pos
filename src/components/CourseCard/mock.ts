export default {
  id: '1',
  caption: 'Especialização em implantodontia',
  href: '/cursos',
  img: 'https://source.unsplash.com/user/willianjusten/200x141'
};
