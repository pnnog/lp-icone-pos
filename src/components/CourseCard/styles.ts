import styled, { css } from 'styled-components';
import media from 'styled-media-query';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    border-radius: ${theme.border.radius};
    display: flex;
    flex-direction: column;
    box-shadow: 1.8px 3.6px 1.8px rgba(0, 0, 0, 0.25);
  `}
`;

export const ImageBox = styled.figure`
  min-height: 14rem;
  width: 100%;
  background: #f6f7f8;
  background-image: linear-gradient(
    to right,
    #f6f7f8 0%,
    #edeef1 20%,
    #f6f7f8 40%,
    #f6f7f8 100%
  );
  background-size: 80rem 14rem;
  animation: placeholderShimmer 1s linear infinite forwards;
  @keyframes placeholderShimmer {
    0% {
      background-position: -40rem 0;
    }
    100% {
      background-position: 40rem 0;
    }
  }
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    display: block;
    border-top-left-radius: ${theme.border.radius};
    border-top-right-radius: ${theme.border.radius};
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    text-align: center;
    padding: ${theme.spacings.xsmall} 0;
    background-color: ${theme.colors.secondary};
    border-bottom-left-radius: ${theme.border.radius};
    border-bottom-right-radius: ${theme.border.radius};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
      padding: ${theme.spacings.small} 0;
    `}
  `}
`;

export const LinkTitle = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    font-weight: ${theme.font.bold};
    font-size: ${theme.font.sizes.small};
    display: inline-block;
    margin: auto;
    max-width: 13.6rem;
    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xlarge};
      padding: ${theme.spacings.small} 0;
      max-width: 22.5rem;

    `};
  `}
`;
