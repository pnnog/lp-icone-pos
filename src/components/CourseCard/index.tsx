import Link from 'next/link';
import * as S from './styles';

export type CourseCardProps = {
  caption: string;
  img: string;
  href: string;
};

const CourseCard = ({ caption, href, img }: CourseCardProps) => (
  <S.Wrapper>
    <Link href={href}>
      <S.ImageBox>
        <S.Image src={img} alt={caption} />
      </S.ImageBox>

      {!!caption && (
        <S.Content>
          <S.Caption> {caption} </S.Caption>
        </S.Content>
      )}
    </Link>
  </S.Wrapper>
);

export default CourseCard;
