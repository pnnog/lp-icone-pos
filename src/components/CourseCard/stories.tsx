import { Meta, Story } from '@storybook/react';
import CourseCard, { CourseCardProps } from '.';
import MockCourseCard from './mock';

export default {
  title: 'CourseCard',
  component: CourseCard,
  args: {
    ...MockCourseCard
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta;

export const Mobile: Story<CourseCardProps> = (args) => (
  <CourseCard {...args} />
);

Mobile.parameters = {
  viewport: {
    defaultViewport: 'mobile1'
  }
};
