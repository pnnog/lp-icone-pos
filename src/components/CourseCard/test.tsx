import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helper';
import MockCourseCard from './mock';

import CourseCard from '.';

describe('<CourseCard />', () => {
  it('should render CourseCard', () => {
    const { container } = renderWithTheme(<CourseCard {...MockCourseCard} />);
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      MockCourseCard.href
    );

    expect(
      screen.getByRole('img', { name: MockCourseCard.caption })
    ).toHaveAttribute('src', MockCourseCard.img);

    expect(container.firstChild).toMatchSnapshot();
  });
});
