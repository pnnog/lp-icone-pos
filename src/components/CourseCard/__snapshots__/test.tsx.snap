// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseCard /> should render CourseCard 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  border-radius: 0.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 1.8px 3.6px 1.8px rgba(0,0,0,0.25);
}

.c1 {
  min-height: 14rem;
  width: 100%;
  background: #f6f7f8;
  background-image: linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100% );
  background-size: 80rem 14rem;
  -webkit-animation: placeholderShimmer 1s linear infinite forwards;
  animation: placeholderShimmer 1s linear infinite forwards;
}

.c2 {
  width: 100%;
  height: 100%;
  display: block;
  border-top-left-radius: 0.4rem;
  border-top-right-radius: 0.4rem;
}

.c3 {
  text-align: center;
  padding: 1.6rem 0;
  background-color: #0284C7;
  border-bottom-left-radius: 0.4rem;
  border-bottom-right-radius: 0.4rem;
}

.c4 {
  color: #F8FAFC;
  font-weight: 600;
  font-size: 1.4rem;
  display: inline-block;
  margin: auto;
  max-width: 13.6rem;
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.0rem;
    padding: 2.4rem 0;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 2.0rem;
    padding: 2.4rem 0;
    max-width: 22.5rem;
  }
}

<article
  class="c0"
>
  <figure
    class="c1"
  >
    <img
      alt="Especialização em implantodontia"
      class="c2"
      src="/img/pic1"
    />
  </figure>
  <div
    class="c3"
  >
    <a
      class="c4"
      href="#"
    >

      Especialização em implantodontia

    </a>
  </div>
</article>
`;
